docs/editor.md

```diff
@@ -103,6 +103,31 @@ cellSpacing = {
 
 [Look at this example to see how cellSpacing affects the layout](//demo/examples/cellSpacing)
 
+### `uiTranslator`
+
+Takes a function `(label?:string) => string | undefined`.  
+
+This prop enables i18n support. All interface labels are wrapped with this function.
+
+Below is the `uiTranslator` function provided in [the i18n demo](//demo/i18n).
+
+```typescript
+const TRANSLATIONS: { [key: string]: string } = {
+  'Edit things': '编辑',
+  'Add things': '添加',
+  'Move things': '移动',
+  'Resize things': '调整大小',
+  'Preview result': '预览模式',
+};
```

Comment:
it would be good to have a type with all the translation keys.

Reply:
The current structure with English words as the key is the laziest and simplest implement. Simplicity is also the main benefit of such design. It hands over 100% of responsibility to lib users and depending on requirements. The dictionary can be as partial or as complete as they want. 

Providing a complete dictionary for the lib users to translate is better, but it requires deep integrations. I have seen other libs do exactly this, most of them not bothering to provide any context descriptions, maybe because interface dictionaries are usually very small. Maybe later when there are contributions of dictionaries, react-page can switch to this solution.

As for using English words as key, I think it is only beneficial for dedicated keys when the same English words require different translations, if not happening, then separate keys posts no benefit because the English word and its translation pairs are unique. Of cause, unless I missed something.  I leave this to your good judgement.

I agree that add an extra description for describing the context could be beneficial. But with the current structure, many of them will be identical. For example, the labels for the buttons of Slate plugins, all labels go through the same render component `PluginButton`, context description needs to be added at a lower level in the plugin code. It means much deeper integrations for complete coverage.

How about changing the signature of `uiTranslator` to `(key: string, description?: string) => string`. The description will be given in English as the context in the source code as needed so that developers can catch it at their end. However, I think most of the labels are very simple and require no description.

Your existing translations object would require developers to provide it at every component as a prop, very performant and easy to use, but require a lot of work both for you and developers. If you agree with the current solution, I think removing `defaultTranslations` is the next step.

Reply:
Maybe in the label, default to English words instead of solely rely on the `t` function. So that you have the flexibility to change labels and not worry about keys being changed.

Reply:
fully agree! let's start simple. If users need more context, let's add something like the proposed `description` prop.